os-daq-modules/daq_rswitch_offload.c

```diff
@@ -1384,6 +1388,35 @@ static void *cleanup_thread(void *ptr)
 	}
 }
 
+static void get_expired_timeout(struct rswitch_context* context)
+{
+	int sz;
+	int fd1 = open(RSWITCH_LIBDAQ_CONF_PATH, O_RDONLY);
+	char cfg[1024];
+	char *ptr;
+
+	if (fd1 < 0) {
+		context->expired_time = ENTRY_TIMEOUT_SEC;
+		return;
+	}
+
+	sz = read(fd1, cfg, 1024);
+	if (sz < strlen(RSWITCH_EXP_TIME_DELIM)) {
+		context->expired_time = ENTRY_TIMEOUT_SEC;
+		return;
+	}
+
+	ptr = strstr(cfg, RSWITCH_EXP_TIME_DELIM);
+	if (!ptr) {
+		context->expired_time = ENTRY_TIMEOUT_SEC;
+	} else {
+		ptr += strlen(RSWITCH_EXP_TIME_DELIM);
```

Comment:
I guess it can brake if the "blacklist_expired_timeout: " key would be exactly at the end of a string

Reply:
I've added initialization with zero values buffer to prevent any issues. In case that you described, atoi, in this case, will return 0 and after that, the default value will be applied.